src/pages/index.js

```diff
@@ -4,42 +4,145 @@ import FormValidator from "../components/FormValidator.js";
 import UserInfo from "../components/UserInfo.js";
 import PopupWithForm from "../components/PopupWithForm.js";
 import Section from "../components/Section.js";
+import Api from "../components/Api.js";
 import PopupWithImage from "../components/PopupWithImage.js";
+import PopupDelete from "../components/PopupDelete.js";
 import { data } from "autoprefixer";
 import {
-  initialCards,
   profileAddButton,
   profileEditButton,
+  avatarEditButton,
+  profileSaveButton,
+  cardCreateSaveButton,
+  avatarSaveButton,
   config,
+  newName,
+  newJob,
   formData,
+  cardElement,
 } from "../Utility/Constant.js";
 
-function handleProfileFormSubmit(formData) {
-  userInfo.setUserInfo(formData);
-  profileEditPopup.close();
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "266f9e9c-f310-4db4-b30c-94bce5a357d4",
+    "Content-Type": "application/json",
+  },
+});
+
+let section;
+let userInfo;
+
+api.userInfoAndCards();
+function createCard(cardData) {
+  const card = new Card(
+    cardData,
+    "#cards-template",
+    handleImageClick,
+    handleCardDelete,
+    handleCardLike
+  );
+  return card.generateCard();
 }
+api.getInitialCards().then((data) => {
+  section = new Section({ items: data, renderer: createCard }, ".cards__list");
+  section.renderItems();
+});
+api.getUser().then((data) => {
+  userInfo = new UserInfo(
+    ".profile__title",
+    ".profile__description",
+    ".profile__image"
+  );
+  userInfo.setUserInfo({ title: data.name, description: data.about });
+  userInfo.changeAvatarImage(data.avatar);
+});
 
-function handleProfileFormCreate(inputValues) {
-  const cardElement = createCard(inputValues);
-  section.addItem(cardElement);
-  profileCardPopup.close();
+function handleProfileFormSubmit(formData) {
+  profileSaveButton.textContent = "Saving...";

```

Comment:
**Could be improved**

It’s better to make method `renderLoading` inside class `PopupWithForm`. You should find the button once in the constructor and change the text:  

```js
  constructor({ popupSelector, handleSubmit }) {
    super(popupSelector);
    // fix the initial button text only once in the constructor
    this._submitBtnText = this._submitBtn.textContent
  }
  // add 2 params: isLoading and loadingText with a default text
  renderLoading(isLoading, loadingText='Saving...') {
    if (isLoading) {
      this._submitBtn.textContent = loadingText;
    } else {
	// here we return back the initial text. So, you don’t need to bother yourself about it
      this._submitBtn.textContent = this._submitBtnText;
    }
  }

```